### ISSUE TITLE
Need ability to include/run one pipeline within another

### ISSUE THREAD

**Issue by brimoor** [voxel51/eta#126]
Options:
(A) support this only at the pipeline metadata level by adding a "pipelines" field that allows access to I/O of other pipelines. When a pipeline is built, a single pipeline config would be populated based on this information
(B) support this at the pipeline config level by allowing pipeline configs to point to other pipeline configs.

I'm leaning towards (A).

**Comment by jasoncorso** [comment]
How does the notion of parallelism impact your decision?

**Comment by brimoor** [comment]
I would like to support parallelism at the module level: the pipeline runner inspects the execution graph and asynchronously runs modules as soon as the necessary inputs are available.

This is realizable regardless of (A) vs (B) above.

**Comment by jasoncorso** [comment]
That's fine then.

Then I'd say in order to decide on (A) or (B) we need concrete examples of
the user-experience differences.  I can envision, for example, that (B)
will be much easier from a user point of view.  For example, we have a
pipeline that handles all the video preprocessing we want (to get it down
to size and rate, etc), then I would very much like to be able to basically
inject it in an anlaytics pipeline much in the same way I would inject a
module.  In other words, I think the user experience should basically be
that (to a pipeline developer) a pipeline is just another type of module to
drop into a new pipeline.

**Comment by brimoor** [comment]
I agree and (A) makes it trivial to do just that. If I want to include the video preprocessing pipeline at the beginning of my new pipeline, I just reference it in the metadata file of my new pipeline and it's there.

If a pipeline developer is using (B), that means they're still writing their own pipeline/module configs manually, and that's bad. Pipeline builder should do everything!

**Comment by jasoncorso** [comment]
Then (A) sounds like the answer.  (I did not understand what (A) and (B)
were from the original description.)
